apps/api/src/app/benchmark/benchmark.service.ts

```diff
@@ -217,6 +224,17 @@ export class BenchmarkService {
   } & UniqueAsset): Promise<BenchmarkMarketDataDetails> {
     const marketData: { date: string; value: number }[] = [];
 
+    const endDate = new Date();
+    const days = differenceInDays(endDate, startDate) + 1;
+    const step = Math.round(days / Math.min(days, MAX_CHART_ITEMS));
+    const dates = Array.from({ length: days / step }).map((_, i) =>
```

Comment:
Could you use `forEach()` instead of `map()`?

Reply:
Replaced with `eachDayOfInterval` :+1: